tests/test_web_api.py

```diff
@@ -0,0 +1,307 @@
+from hashlib import md5
+from mock import patch
+from os import path as ospath
+from regparser.web.jobs.models import job_status_values
+from regparser.web.jobs.utils import (
+    eregs_site_api_url,
+    file_url,
+    status_url
+)
+from regparser.web.jobs.views import FileUploadView as PatchedFileUploadView
+from rest_framework.test import APITestCase
+from tempfile import NamedTemporaryFile
+try:
+    from urllib.parse import urlparse
+except ImportError:
+    from urlparse import urlparse
+
+fake_pipeline_id = "f726e1e0-d43c-4eb7-9274-681ddaba427a"
+fake_email_id = "4774f0f6-b53e-4b34-821e-fc8ae5e113fe"
+
+
+def fake_add_redis_data_to_job_data(job_data):
+    return job_data
+
+
+def fake_delete_eregs_job(job_id):
+    return None
+
+
+def fake_redis_job(args, timeout=60*30, result_ttl=-1):
+    return type("FakeRedisJob", (object, ), {"id": fake_pipeline_id})
+
+
+def fake_queue_email(job, statusurl, email_address):
+    return fake_email_id
+
+
+# Even though these functions are in the ``utils`` module, the code paths we're
+# following mean we're encountering them in the ``views`` namespace:
+@patch("regparser.web.jobs.views.add_redis_data_to_job_data",
+       fake_add_redis_data_to_job_data)
+@patch("regparser.web.jobs.views.delete_eregs_job", fake_delete_eregs_job)
+@patch("regparser.web.jobs.views.queue_eregs_job", fake_redis_job)
+@patch("regparser.web.jobs.views.queue_notification_email", fake_queue_email)
```

Comment:
Not a big deal for now, but needing to mock out many different modules/libraries is a code smell as it implies many dependencies. Ideally we'd be able to refactor this to a single `Mock`